File: Java/civitas/src/civitas/Especulador.java
package civitas;


public class Especulador extends Jugador{
    private int fianza;
    private static int factorEspeculador = 2;
    private boolean soyEspe=false;
    
    Especulador(Jugador _jugador, int _fianza){
        super(_jugador);
        fianza = _fianza;
        soyEspe=true;
        for(int i=0; i<super.getPropiedades().size(); i++){
            _jugador.getPropiedades().get(i).actualizaPropietarioPorConversion(this);
        }
    }
    
    @Override
    boolean encarcelar(int _casillaCarcel){
        boolean resul=false;
        if(super.encarcelado==false){
            if(super.tieneSalvoconducto()==false){
                if(fianza>=super.getSaldo()){
                    resul = true;
                }
                else{
                    super.paga(fianza);
                    Diario.getInstance().ocurreEvento("El jugador " + super.getNombre() + "paga la fianza");
                }
            }
            else{
                resul=false;
                super.perderSalvoconducto();
                Diario.getInstance().ocurreEvento("El jugador " + super.getNombre() + "sale de la carcel por salvoconducto");
            }
        }  
        return resul;
    }
    @Override
    public boolean soyEspeculador(){
        return true;
    }
    
    @Override
    public int getCasasMax(){
        return(super.getCasasMax()*factorEspeculador);
    }
    
    @Override
    public int getHotelesMax(){
        return(super.getHotelesMax()*factorEspeculador);
    }
    
    @Override
    public boolean pagaImpuesto(float _cantidad){
        boolean resul = false;
        if(encarcelado==false){
            resul = paga(_cantidad/2);
            Diario.getInstance().ocurreEvento("El jugador " + super.getNombre() + " paga " + Float.toString(_cantidad/2) + " por un impuesto");
        }
        return resul;
    }
    
    @Override
    public String toString(){
        String texto="";
        texto= "Jugador { Nombre: "+ super.getNombre() + " ,saldo: " + super.getSaldo();
        if(encarcelado)
            texto+=", estoy encarcelado.";
        else
            texto+=", no estoy encarcelado ";
        texto+=" y soy un Especulador";
        return texto;
    }
    
}

File: Ruby/civitas/pkg/civitas-0.0.1/lib/especulador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

class Especulador < Jugador
  @@factor_especulador = 2
  def initialize(_jugador, _fianza)
    super(_jugador)
    @fianza=_fianza
    @soy_espe=true
    for i in (super.propiedades.size()-1)
      _jugador.propiedades.at(i).actualizar_propietario_por_converision(self)
    end
  end
  
  def encarcelar(_casilla)
    resul = false
    if(super.encarcelado==false)
      if(super.tiene_salvoconducto==false)
        if(@fianza>=super.saldo)
          resul=true
        else
          super.paga(@fianza)
        end
      else
        resul=false
        super.perder_salvoconducto()
        Diario.Instance.ocurre_evento("Jugador pierde salvo")
      end
    end
    return resul
  end
  
  def soy_especulador()
    return @soy_espe
  end
  
  def get_casas_max()
    return (super.casas_max*@@factor_especulador)
  end
  
  def get_hoteles_max()
    return (super.hoteles_max*@@factor_especulador)
  end
  
  def pagar_impuesto(_cantidad)
    resul = false
    if(super.encarcelado==false)
      resul = super.paga(_cantidad/2)
      Diario.Instance.ocurre_evento("Pagas impuesto")
    end
    return resul
  end
  
  def to_sj()
      texto = "Jugador { Nombre: #{super.nombre}, saldo: #{super.saldo} "
      if(super.encarcelado)
        texto+=" y estoy encarcelado"
      else
        texto+=" y no estoy encarcelado"
      end
      texto+=" y SOY un especulador "
      
    
      return texto
    end
end
